src/Features/Core/Portable/CodeFixes/AddExplicitCast/AbstractAddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,403 @@
+        /// <param name="targetNode"> The node needs to be cast.</param>
+        /// <returns>
+        /// Return true if the invocation expression with new arguments is applicatble.
+        /// Otherwise, return false
+        /// </returns>
+        protected bool IsInvocationExpressionWithNewArgumentsApplicable(
+            SemanticModel semanticModel, SyntaxNode root, SyntaxNode oldArgumentList,
+            List<SyntaxNode> newArguments, SyntaxNode targetNode)
+        {
+            var newRoot = root.ReplaceNode(oldArgumentList, GenerateNewArgumentList(oldArgumentList, newArguments));
+            if (newRoot.FindNode(targetNode.Span).GetAncestorOrThis<TAttributeSyntax>() is TAttributeSyntax attribute)
+            {
+                var symbolInfo = GetSpeculativeAttributeSymbolInfo(semanticModel, attribute.SpanStart, attribute);
+                return symbolInfo.Symbol != null;
+            }
+            else if (newRoot.FindNode(targetNode.Span).GetAncestorOrThis<TArgumentListSyntax>() is TArgumentListSyntax argumentList)
+            {
+                if (argumentList.Parent is SyntaxNode newNode)
+                {
+                    var symbolInfo = semanticModel.GetSpeculativeSymbolInfo(
+                        newNode.SpanStart, newNode, SpeculativeBindingOption.BindAsExpression);
+                    return symbolInfo.Symbol != null;
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Collect all the available cast pairs, format is (target argument expression, potential conversion type)
+        /// </summary>
+        /// <param name="targetArgument"> The argument that need to be cast</param>
+        /// <param name="argumentList"> The argument list that contains the target argument to be cast </param>
+        /// <param name="invocationNode"> The invocation node that is the parent of "argumentList"</param>
+        /// <returns>
+        /// Return all the available cast pairs, format is (target argument expression, potential conversion type)
+        /// </returns>
+        protected ImmutableArray<(TExpressionSyntax, ITypeSymbol)> GetPotentialConversionTypes(
+            ISyntaxFactsService syntaxFacts, SemanticModel semanticModel, SyntaxNode root, SyntaxNode targetArgument,
+            SyntaxNode argumentList, SyntaxNode invocationNode, CancellationToken cancellationToken)
+        {
+            // Implicit downcast appears on the argument of invocation node,
+            // get all candidate functions and extract potential conversion types 
+            var symbolInfo = semanticModel.GetSymbolInfo(invocationNode, cancellationToken);
+            using var _ = ArrayBuilder<ISymbol>.GetInstance(out var candidateSymbols);
+            candidateSymbols.AddRange(symbolInfo.CandidateSymbols);
+            if (symbolInfo.Symbol != null) // BC42016: the only candidate symbol is symbolInfo.Symbol
+                candidateSymbols.Add(symbolInfo.Symbol);
```

Comment:
```suggestion
            if (symbolInfo.Symbol != null) // BC42016: the only candidate symbol is symbolInfo.Symbol
                candidateSymbols.Add(symbolInfo.Symbol);
            else
                candidateSymbols.Add(symbolInfo.CandidateSymbols)
```

Reply:
Assuming I understood this comment right, if the symbol is found then candidatesymbols won't matter?

Reply:
yeah true